### PR TITLE
Fix a race condition in parallel table loading.

### PR DESCRIPTION
**Pull request by akuzm**
ThreadSanitizer reported that a data race occurs when we attach to a
database and load its tables concurrently. Remove the custom code that
waits for all tables to load, and replace it with a local thread pool
and its wait() method.

This changes observable behavior: before, we would load the tables
sequentially when loading the system database, or performing an ATTACH
DATABASE query. Now we always load tables of ordinary databases in
parallel.

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Build/Testing/Packaging Improvement

### CODE REVIEW

File: dbms/src/Core/Defines.h [hunk 1]
```diff
@@ -139,3 +139,7 @@
 /// This number is only used for distributed version compatible.
 /// It could be any magic number.
 #define DBMS_DISTRIBUTED_SENDS_MAGIC_NUMBER 0xCAFECABE
+
+/// A macro for suppressing warnings about unused variables. Useful for
+/// structured bindings which have no standard way to declare this.
+#define UNUSED_VARIABLE(X) (void) (X)
```
Comment (alexey-milovidov) [inline dbms/src/Core/Defines.h]: This works not only for variables, but also for function results.
So, it's Ok to name it `UNUSED`.

**Review by alexey-milovidov — APPROVED** [review]
That's great!

And it is the natural solution after we introduced global thread pool.

**Comment by alexey-milovidov** [comment]
PS. Probably it may create excessive CPU load in presence of many databases?

**Comment by akuzm** [comment]
> PS. Probably it may create excessive CPU load (number of threads waiting for CPU) in presence of many databases?

Databases are loaded sequentially, and the number of threads per database is capped with the number of physical cores (as before), so it's probably OK.

**Pull request merged**